src/CBLDatabase_Internal.hh

```diff
@@ -141,16 +128,18 @@ public:
     
     
     fleece::MutableArray scopeNames() const {
+        auto db = _c4db->useLocked();
         auto names = FLMutableArray_New();
-        _c4db->useLocked()->forEachScope([&](slice scope) {
+        db->forEachScope([&](slice scope) {
             FLMutableArray_AppendString(names, scope);
         });
         return names;
```

Comment:
Why is this not simply c4db_scopeNames?

Reply:
I don't want to mix C API call into the implementation while the other places are all using C++ API.

Reply:
This made me think that why C++ API doesn't have a function that returns an array?